roachtest: clean up the canary tests

BramGruneir opened the pull request:
Move all the repeat logic into canary.go and make each step have it's own
repeat loop.
Also DRY up a the blacklist/skiplist logic.

AND ensure that the canary tests all use the latest release instead of a
hardcoded version.  This will give us a head's up if something has gone wrong
with their release.

Fixes #35979.

Release note: None

BramGruneir committed:
roachtest: clean up the canary tests

Move all the repeat logic into canary.go and make each step have it's own
repeat loop.
Also DRY up a the blacklist/skiplist logic.

AND ensure that the canary tests all use the latest release instead of a
hardcoded version.  This will give us a head's up if something has gone wrong
with their release.

Fixes #35979.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/36107)
<!-- Reviewable:end -->

BramGruneir commented:
This replaces the erroneously closed #36037.

BramGruneir committed:
roachtest: fix the TypeORM test

The s3 binary for sqlite3 is no longer available so it must be built from
source. Note that it's not used in the test directly, but it considered a
dependency to install TypeORM and run the unit tests.

Fixes #36074

Release note: None

BramGruneir commented:
Added a fix to the TypeORM test, since I'd rather not have to go back and merge the changes it requires manually.

BramGruneir commented:
TFTR!

bors r+
